File: frontend/src/types/inventoryItem.ts
export interface InventoryItemValue {
  id: string;
  field_id: string;
  value: unknown;
  created_at: string;
  updated_at: string;
}

export interface InventoryItem {
  id: string;
  inventory_id: string;
  values: InventoryItemValue[];
  created_at: string;
  updated_at: string;
}

export interface InventoryItemValueInput {
  field_id: string;
  value: unknown;
}

export interface InventoryItemInput {
  values: InventoryItemValueInput[];
}

File: backend/app/schemas/inventory_item.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class InventoryItemValueInput(BaseModel):
    field_id: str
    value: Any


class InventoryItemCreate(BaseModel):
    values: list[InventoryItemValueInput] = Field(default_factory=list)


class InventoryItemUpdate(BaseModel):
    values: list[InventoryItemValueInput] = Field(default_factory=list)


class InventoryItemValueRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    field_id: str
    value: Any
    created_at: datetime
    updated_at: datetime


class InventoryItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    inventory_id: str
    values: list[InventoryItemValueRead] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime
